### PR TITLE
AOP2-69: Change role

### PR DESCRIPTION
**Pull request by waleedusman**

### CODE REVIEW

File: backend/app/api/user.py [hunk 1]
```diff
@@ -149,9 +149,10 @@ def update_user_details(**kwargs):
     if kwargs.get('email_address', None) is not None:
         user.email_address = kwargs['email_address']
     if kwargs.get('role', None) is not None:
-        if user.role == 'supplier' and kwargs['role'] != user.role:
-            user.supplier_code = None
-            kwargs.pop('supplierCode', None)
+        if (kwargs['role'] not in ['buyer', 'primary-user']) or\
```
Comment (qimyi) [inline backend/app/api/user.py]: why not give kwargs['role'] a name? 
```
targetRole = kwargs['role']
if targetRole is not None...
```

File: frontend/apps/orams/actions/adminSearchActions.js [hunk 2]
```diff
@@ -130,6 +132,23 @@ const updateUser = (url, errorMessage, successMessage) => dispatch => {
   })
 }
 
+const updateRole = (url, data, errorMessage, successMessage) => dispatch => {
```
[the remark this answers is not in the document]
Comment (qimyi) [reply frontend/apps/orams/actions/adminSearchActions.js]: you can refactor `updateRole` to take `data` as an argument
```
const updateUser = (url, data, errorMessage, successMessage) => dispatch => {
```

but all the other place that calls updateUser needs to be updated to pass a second argument. The good news is updateUser is not exported and only used in this file. 
An easier way is to add the `data` argument as the last item in the parameter list. That way, you don't need to change the way it's called in other places, but IMO not as nice in terms of parameter ordering

File: frontend/apps/orams/actions/adminSearchActions.js [hunk 3]
```diff
@@ -17,7 +17,9 @@ import {
   ACTIVATE_USER_ERROR,
   ACTIVATE_USER_SUCCESS,
   UNLOCK_USER_ERROR,
-  UNLOCK_USER_SUCCESS
+  UNLOCK_USER_SUCCESS,
```
Comment (qimyi) [inline frontend/apps/orams/actions/adminSearchActions.js]: Nothing to do with this line of code. How do you make a comment on a file level? 
Anyway, I think the file name does not match the intent of this file anymore. Probably add that to tech debt to refactor.

File: frontend/apps/orams/actions/adminSearchActions.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (waleedusman) [reply frontend/apps/orams/actions/adminSearchActions.js]: done

File: backend/app/api/user.py (hunk 1 above)
Comment (waleedusman) [reply backend/app/api/user.py]: i didn't think it was needed here given the size of the method and what's going on in it

File: frontend/apps/orams/actions/adminSearchActions.js [hunk 4]
```diff
@@ -114,10 +116,11 @@ export const loadUserProfileData = id => dispatch => {
   })
 }
 
-const updateUser = (url, errorMessage, successMessage) => dispatch => {
+const updateUser = (url, data = {}, errorMessage, successMessage) => dispatch => {
```
Comment (bsigney) [inline frontend/apps/orams/actions/adminSearchActions.js]: setting the data as second param is not going to work. the other functions calling `updateUser` will pass on their second param into `data` here.

I think your best bet is to move the `data` param to the end of param list

File: frontend/apps/orams/components/UserProfile/UserProfile.js [hunk 5]
```diff
@@ -93,11 +100,45 @@ class UserProfile extends Component {
     return ''
   }
 
+  renderUpdateRoleButton(userProfileData) {
+    const { id, role } = userProfileData
+    if (role[0] === 'buyer') {
```
[the remark this answers is not in the document]
Comment (bsigney) [reply frontend/apps/orams/components/UserProfile/UserProfile.js]: yes for the `/user/:id` endpoint.  Though not sure why it is returning an array of roles - something we inherited from DM

File: frontend/apps/orams/actions/adminSearchActions.js (hunk 4 above)
Comment (waleedusman) [reply frontend/apps/orams/actions/adminSearchActions.js]: thanks for spotting that! I was wondering why the success/failure message were not being displayed when toggling activate/deactivate. I'll update the other invocations in this file to pass an empty body.

**Pull request merged**